### ISSUE TITLE
Hey i have been testing and great product but 1 huge issue i am running into. 

### ISSUE THREAD

**Issue by nobonesjones** [stackblitz/bolt.new#242]
**Is your feature request related to a problem? Please describe:** Sometimes, your work will be totally redone, removed, or deleted even if you aren't working on that page. e.g., spent time on 2 dashboards, then finished and moved to a simple landing page to finish off. But somehow, making the landing page wiped the 2 dashboards. There needs to be a version control or an undo button, or....maybe it just shouldn't remove things!!

 something not related to my prompt totally changes and breaks. 

**Describe the solution you'd like:**

**Describe alternatives you've considered:**

**Additional context:**

**Comment by UltraInstinct0x** [comment]
I encountered a similar issue where the bolt became stuck on “creating …js.” Upon reloading the page, most of the recently made changes (despite their visible rendering in the preview) vanished, and the package.json file was corrupted. Consequently, I had to redo a significant portion of my work, incurring substantial costs.
The outcome of my efforts was unsatisfactory, especially considering the limited investment of $10. Perhaps I could have provided better prompting. Nevertheless, the issue of getting stuck should be resolved. Additionally, the implementation of an option to export all files would be beneficial in preventing such errors.
